service/auth/service.go

```diff
@@ -71,5 +76,22 @@ func (s *Service) Unregister(id string) error {
 		return fmt.Errorf("unregister failed: %w", err)
 	}
 
+	// Invalidate token when unregistering
+	s.sessionCache.Delete(id)
+
 	return nil
 }
+
+func (s *Service) Login(id, key string) (string, error) {
+	if err := s.Authenticate(id, key); err != nil {
+		return "", fmt.Errorf("login failed: %w", err)
+	}
+	bearerToken, err := s.sessionCache.NewUniqueToken()
```

Comment:
If we are using it like this I am thinking we may as well decouple it completely from the cache as it becomes more of a utility function that can safely live in the `auth` package. What do you think?

Reply:
That could be done but it's not guaranteed anymore that the new token is unique then.